### PR TITLE
Add Amazon ECS plugin

### PR DESCRIPTION
**Pull request by ChazEpps**
Please review

### CODE REVIEW

File: src/main/java/jp/classmethod/aws/gradle/ecs/JsonUnmarshallerContextHelper.java [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+import com.amazonaws.transform.SimpleTypeJsonUnmarshallers.ShortJsonUnmarshaller;
+import com.amazonaws.transform.SimpleTypeJsonUnmarshallers.StringJsonUnmarshaller;
+import com.amazonaws.transform.Unmarshaller;
+import com.amazonaws.util.ImmutableMapParameter;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+
+@SuppressWarnings("PMD")
```
Comment (dai0304) [inline src/main/java/jp/classmethod/aws/gradle/ecs/JsonUnmarshallerContextHelper.java]: Don't suppress all PMD warnings.
And would you describe why do you think this suppression is reasonable?

Comment (ChazEpps) [inline src/main/java/jp/classmethod/aws/gradle/ecs/JsonUnmarshallerContextHelper.java]: Done.  I have inevitably copied some code from Amazon. So some violations occurred.
Thank you!

**Comment by ChazEpps** [comment]
@dai0304  Ready to review 👍

**Comment by mochizuki-masao** [comment]
@ChazEpps Sorry for the delay of reviewing. checked.
@dai0304 I reviewed the code and checked some of the PR tasks, not whole tasks, and seemed to work well.

**Comment by ChazEpps** [comment]
@mochizuki-masao Oh you did not forget. Thank you.

**Comment by mochizuki-masao** [comment]
@ChazEpps 

I discussed with @dai0304 and one question about this PR.
You said you copied this source code from Amazon. Can you show where the original source code is?
I'm worrying about the License of this code is proper, redistributable one.
https://github.com/classmethod/gradle-aws-plugin/pull/103/files/9c3697403afa6a90bf69d471af2eaba48ae6675c#diff-3e1d55f6e55b61317bea77ebe2b99609

Even if the above question is solved, still we are looking for more simple resolution for deserializing the task definition and service JSON.
I know the ECS data model is very complicated and hard to model into Gradle task. We would like to try to keep this plugin simple, but no good idea, currently...

**Comment by dongjun91** [comment]
@mochizuki-masao
I contacted AWS Support and I received below. 

> Hi,
> 
> Thanks for contacting AWS Support.
> 
> With regard to the licensing of the our open source Java SDK software, as it's under the Apache 2.0 License as outlined here:
> https://github.com/aws/aws-sdk-java/blob/master/LICENSE.txt
> as long as you follow the Terms and Conditions listed in this License document, in this case more specifically for Step 4. Redistribution(though the whole thing should be adhered to), you will be within the usage of the license.
> 
> In your case as you mentioned that you would like to use this for your own Open-Source project if you ensure that the points listed under Step 4. are meet, which essentially comes down to ensuring the NOTICE.txt here:
> https://github.com/aws/aws-sdk-java/blob/master/NOTICE.txt
> is included in your code, that the above LICENSE.txt is also included and that you note any changes that are made you will be within the license for using this Java SDK within your own project.
> 
> As a side note, if you have any specific questions about these sorts of licensing and compliance areas I would recommend using the link here:
> https://aws.amazon.com/contact-us/
> under the sub-section:
> I have a compliance question.
> as that will be directed towards people who deal with these areas on a daily basis.
> 
> That being said, if you have any additional questions about this, please don't hesitate to ask.
> 
> Best regards,
> 
> Jamis M. | Sydney Support Centre
> Amazon Web Services
>

**Comment by dongjun91** [comment]
Seems like using the code should be ok.

**Pull request closed without merging**